platform/linux-generic/pktio/socket_xdp.c

```diff
@@ -106,219 +114,86 @@ static int sock_xdp_init_global(void)
-
-	if (count <= 0) {
-		++sock_info->i_stats[RX_PKT_ALLOC_ERR];
-		return false;
-	}
-
-	fill_q = &sock_info->fill_q;
-
-	if (xsk_ring_prod__reserve(fill_q, count, &start_idx) == 0U) {
-		odp_packet_free_multi(packets, count);
-		++sock_info->i_stats[RX_DESC_RSV_ERR];
-		return false;
-	}
-
-	pktio_idx = sock_info->pktio_idx;
-	block_size = pool->block_size;
-
-	for (int i = 0; i < count; ++i) {
-		pkt_hdr = packet_hdr(packets[i]);
-		pkt_hdr->ms_pktio_idx = pktio_idx;
-		*xsk_ring_prod__fill_addr(fill_q, start_idx++) =
-			pkt_hdr->event_hdr.index.event * block_size;
-	}
-
-	xsk_ring_prod__submit(&sock_info->fill_q, count);
-
-	return true;
-}
-
-static inline void lock_rxtx(pkt_xdp_t *priv)
-{
-	odp_ticketlock_lock(&priv->rx_lock);
-	odp_ticketlock_lock(&priv->tx_lock);
-}
-
-static inline void unlock_rxtx(pkt_xdp_t *priv)
-{
-	odp_ticketlock_unlock(&priv->rx_lock);
-	odp_ticketlock_unlock(&priv->tx_lock);
+	return (xdp_sock_info_t *)(uintptr_t)(pktio_entry->s.pkt_priv);
 }
 
 static int sock_xdp_stats_reset(pktio_entry_t *pktio_entry)
 {
-	pkt_xdp_t *priv = pkt_priv(pktio_entry);
-
-	lock_rxtx(priv);
 	memset(&pktio_entry->s.stats, 0, sizeof(odp_pktio_stats_t));
-	unlock_rxtx(priv);
 
 	return 0;
 }
 
 static int sock_xdp_open(odp_pktio_t pktio, pktio_entry_t *pktio_entry, const char *devname,
 			 odp_pool_t pool_hdl)
 {
-	pkt_xdp_t *priv;
+	xdp_sock_info_t *priv;
 	pool_t *pool;
-	struct xsk_socket_config config;
-	uint32_t bind_q;
 	int ret;
 
 	if (disable_pktio)
 		return -1;
 
 	priv = pkt_priv(pktio_entry);
-	memset(priv, 0, sizeof(pkt_xdp_t));
+	memset(priv, 0, sizeof(xdp_sock_info_t));
 	pool = pool_entry_from_hdl(pool_hdl);
-	priv->sock_info.umem_info = (xdp_umem_info_t *)pool->mem_src_data;
-	priv->sock_info.xsk = NULL;
+	priv->umem_info = (xdp_umem_info_t *)pool->mem_src_data;
 	/* Mark transitory kernel-owned packets with the pktio index, so that they can be freed on
 	 * close. */
-	priv->sock_info.pktio_idx = 1 + odp_pktio_index(pktio);
-	fill_socket_config(&config);
-	bind_q = get_bind_queue_index(devname);
-	/* With xsk_socket__create_shared(), as only one bind queue index can
-	 * be passed, NIC in use needs to be configured accordingly to have
-	 * only a single combined TX-RX queue, otherwise traffic may not end up
-	 * on the socket. For now, always bind to the first queue (overridable
-	 * with environment variable). */
-	ret = xsk_socket__create_shared(&priv->sock_info.xsk, devname, bind_q,
-					priv->sock_info.umem_info->umem, &priv->sock_info.rx,
-					&priv->sock_info.tx, &priv->sock_info.fill_q,
-					&priv->sock_info.compl_q, &config);
-
-	if (ret) {
-		ODP_ERR("Error creating xdp socket for bind queue %u: %d\n", bind_q, ret);
-		goto xsk_err;
-	}
-
-	/* Ring setup/clean up routines seem to be asynchronous with some drivers and might not be
-	 * ready yet after xsk_socket__create_shared(). */
-	sleep(1U);
-
+	priv->pktio_idx = 1 + odp_pktio_index(pktio);
+	strncpy(priv->devname, devname, IF_NAMESIZE - 1U);
```

Comment:
Looks like `devname` is already saved to `pktio_entry->s.name`.

Reply:
Yes, and it's also used in promisc/link status/mac functions but peculiarly not elsewhere. Used also elsewhere in v3.